juju/machine.py

```diff
@@ -176,15 +176,15 @@ def run(self, command, timeout=None):
         """ Execute an scp command. Requires a fully qualified source and
         destination.
         """
-        cmd = [
-            'scp',
+        cmd = ['scp']
+        cmd += scp_opts.split()
```

Comment:
@ryan-beisner 
In this specific case, `+=` is not used with strings directly but with a list of strings `scp_opts.split()`
So, it would result in something like:
```python
>>> cmd = ['scp']
>>> scp_opts = '-p -r'
>>> cmd += scp_opts.split()
>>> cmd
['scp', '-p', '-r']
```
Which is what is expected.
Nevertheless, I could use `cmd.extend(scp_opts.split())` it you prefer

Reply:
I really don't like the existing interface (`scp_opts` as a `str`), since it means you can't pass in any option values that actually contain a space.  Do you think you could wrap the split in an `if isinstance(scp_opts, str)` and update the docs to indicate that it supports a list or string?